### ISSUE TITLE
Excessive write I/O since upgrade from 0.62/0.63 to 0.69.1

### ISSUE THREAD

**Issue by monty241** [RocketChat/Rocket.Chat#12046]
### Description:

Upgraded using standard steps on t2 instance with 40 GB (120 IOPS SSD) on September 5.

Since then, the write disk I/O has been growing considerably. See pictures.

Waiting I/O is typically now close to 100%. Typically when node is consuming CPU and memory.

Resized instance to 120 GB to get some more IOPS since it was hardly possible to log on.

### Expected behavior:

Similar disk write I/O as before.

### Server Setup Information:

- Version of Rocket.Chat Server:  0.69
- Operating System: Ubuntu, Linux 3.13.0
- Deployment Method: tar
- Number of Running Instances: 1
- DB Replicaset Oplog: ?
- NodeJS Version: v8.9.4
- MongoDB Version: v3.0.15

**Comment by monty241** [comment]
![image](https://user-images.githubusercontent.com/7579444/45508506-4cb8f500-b795-11e8-95f9-c58b2151b585.png)

![image](https://user-images.githubusercontent.com/7579444/45508519-53476c80-b795-11e8-8a10-5f79999db3de.png)

![image](https://user-images.githubusercontent.com/7579444/45508537-5d696b00-b795-11e8-9335-4cb4741a492b.png)

**Comment by monty241** [comment]
There are no cron jobs and at jobs. It seems that RocketChat itself triggers some maintenance jobs around midnight which start the high I/O use period.

**Comment by monty241** [comment]
Especially writes:

![image](https://user-images.githubusercontent.com/7579444/45508663-a7eae780-b795-11e8-958e-65de9d337655.png)

**Comment by geekgonecrazy** [comment]
I assume mongo is on this same server?  This looks like mongo writes

**Comment by monty241** [comment]
Yes, Mongo is on the same server. Checked current status after increasing the IOPS from 120 to 360 (it is a burstable AWS instance). With 360 IOPS (and associated IO credits) it can handle the I/O load before depleting the burst balance.

Picture:
- 1: prolonged period of down (no I/O credits left)
- 2 and 3: did not complete consume all I/O credits, so no impact on production
- 4 and 5: short period of time the I/O credits were all gone, but before users entered the office there were sufficiently new credits allocated.

Depletion seems to start at 11 PM and last till 1 AM UTC (server is also in UTC time zone).

![image](https://user-images.githubusercontent.com/7579444/46127075-d35ae100-c22f-11e8-962f-2558f81e9520.png)

A few days ago we have disabled purging of old messages. It seems that after that the problem is completely gone.

Maybe the purging needs some additional work for performance, but that is just a wild guess.

**Comment by geekgonecrazy** [comment]
Just checking back in.  Has still been good?  How did you have message purging configured?
